src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeValueFormatter.java

```diff
@@ -0,0 +1,133 @@
+                                  CaseViewFieldBuilder caseViewFieldBuilder) {
+        super(caseViewFieldBuilder);
+        this.dateTimeFormatParser = dateTimeFormatParser;
+    }
+
+    @Override
+    protected CaseViewField executeSimple(CaseViewField caseViewField, BaseType baseType) {
+        caseViewField.setFormattedValue(
+            caseViewField.getValue() instanceof TextNode && hasSupportedDisplayContextParameter(caseViewField.getDisplayContextParameter()) ?
+                executeSimple((TextNode) caseViewField.getValue(), caseViewField, baseType, caseViewField.getId(), null, caseViewField) :
+                caseViewField.getValue()
+        );
+
+        return caseViewField;
+    }
+
+    @Override
+    protected CaseViewField executeCollection(CaseViewField caseViewField) {
+        caseViewField.setFormattedValue(
+            caseViewField.getValue() instanceof ArrayNode && hasSupportedDisplayContextParameter(caseViewField.getDisplayContextParameter()) ?
+                executeCollection((ArrayNode) caseViewField.getValue(), caseViewField, caseViewField.getId(), null, caseViewField) :
+                caseViewField.getValue()
+        );
+
+        return caseViewField;
+    }
+
+    @Override
+    protected JsonNode executeSimple(JsonNode node, CommonField field, BaseType baseType, String fieldPath, WizardPageComplexFieldOverride override, CaseViewField topLevelField) {
+        final DisplayContext displayContext = displayContext(topLevelField, override);
+        return !isNullOrEmpty(node)
+            && hasDisplayContextParameterType(field.getDisplayContextParameter(), ENUM_MAP.get(displayContext))
+            && isSupportedBaseType(baseType, SUPPORTED_TYPES) ?
+            createNode(field.getDisplayContextParameter(), node.asText(), baseType, fieldPath, displayContext) :
+            node;
+    }
+
+    @Override
+    protected JsonNode executeCollection(JsonNode collectionNode, CommonField caseViewField, String fieldPath, WizardPageComplexFieldOverride override, CaseViewField topLevelField) {
+        final BaseType collectionFieldType = BaseType.get(caseViewField.getFieldType().getCollectionFieldType().getType());
+        final DisplayContext displayContext = displayContext(topLevelField, override);
+
+        if (hasDisplayContextParameterType(caseViewField.getDisplayContextParameter(), ENUM_MAP.get(displayContext))
+            && isSupportedBaseType(collectionFieldType, SUPPORTED_TYPES)) {
+            ArrayNode newNode = MAPPER.createArrayNode();
+            collectionNode.forEach(item -> {
+                JsonNode newItem = item.deepCopy();
+                ((ObjectNode)newItem).replace(CollectionValidator.VALUE,
+                    createNode(caseViewField.getDisplayContextParameter(), item.get(CollectionValidator.VALUE).asText(), collectionFieldType, fieldPath, displayContext));
+                newNode.add(newItem);
+            });
+
+            return newNode;
+        }
```

Comment:
the logic that loops through collection fields and navigates complex fields is common logic that applies to all processors. So it should be made generic and written only once rather than being inside each single subclass of CaseViewFieldProcessor. But we can ignore this comment for now as I have raised a separate ticket to introduce a generic Processor